File: library/src/main/java/com/zeropercenthappy/tablayoutlibrary/ZPHTabLayoutConfig.java
package com.zeropercenthappy.tablayoutlibrary;

import android.widget.ImageView;

/**
 * @author ybq
 * @date 2017/12/9
 */

public class ZPHTabLayoutConfig {
    private final static int CENTER = 0;
    private final static int CENTER_CROP = 1;
    private final static int CENTER_INSIDE = 2;
    private final static int FIT_CENTER = 3;
    private final static int FIT_END = 4;
    private final static int FIT_START = 5;
    private final static int FIT_XY = 6;
    private final static int MATRIX = 7;

    static void setScaleType(ImageView imageView, int scaleType) {
        switch (scaleType) {
            case CENTER:
                imageView.setScaleType(ImageView.ScaleType.CENTER);
                break;
            case CENTER_CROP:
                imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
                break;
            case CENTER_INSIDE:
                imageView.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
                break;
            case FIT_CENTER:
                imageView.setScaleType(ImageView.ScaleType.FIT_CENTER);
                break;
            case FIT_END:
                imageView.setScaleType(ImageView.ScaleType.FIT_END);
                break;
            case FIT_START:
                imageView.setScaleType(ImageView.ScaleType.FIT_START);
                break;
            case FIT_XY:
                imageView.setScaleType(ImageView.ScaleType.FIT_XY);
                break;
            case MATRIX:
                imageView.setScaleType(ImageView.ScaleType.MATRIX);
                break;
            default:
                imageView.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
                break;
        }
    }
}

File: tablayoutlibrary/src/main/java/com/zeropercenthappy/tablayoutlibrary/ZPHTabLayoutConfig.kt
package com.zeropercenthappy.tablayoutlibrary

import android.widget.ImageView

/**
 * @author ybq
 * @date 2017/12/9
 */

object ZPHTabLayoutConfig {
    private const val CENTER = 0
    private const val CENTER_CROP = 1
    private const val CENTER_INSIDE = 2
    private const val FIT_CENTER = 3
    private const val FIT_END = 4
    private const val FIT_START = 5
    private const val FIT_XY = 6
    private const val MATRIX = 7

    internal fun setScaleType(imageView: ImageView, scaleType: Int) {
        when (scaleType) {
            CENTER -> imageView.scaleType = ImageView.ScaleType.CENTER
            CENTER_CROP -> imageView.scaleType = ImageView.ScaleType.CENTER_CROP
            CENTER_INSIDE -> imageView.scaleType = ImageView.ScaleType.CENTER_INSIDE
            FIT_CENTER -> imageView.scaleType = ImageView.ScaleType.FIT_CENTER
            FIT_END -> imageView.scaleType = ImageView.ScaleType.FIT_END
            FIT_START -> imageView.scaleType = ImageView.ScaleType.FIT_START
            FIT_XY -> imageView.scaleType = ImageView.ScaleType.FIT_XY
            MATRIX -> imageView.scaleType = ImageView.ScaleType.MATRIX
            else -> imageView.scaleType = ImageView.ScaleType.CENTER_INSIDE
        }
    }
}
